File: operaciones.py
print('introduce un numero grande')
num = input()
n= int(len(num))
#suma
sum=int(0)
#multiplicacion 
mult = int(1)
#division contra el ultimo digito diferente de cero
div=0
dividendo=0
d = -1
#resta de impares
resta = 0
sumadeimpares = 0
sumadepares=0
for i in range(n):
    sum = sum + int(num[i])
    mult = mult * int(num[i])
    if num[d] == '0':
        d = d-1
    else:
        dividendo = sum + int(num[i])
        div= dividendo / int(num[d])
    if int(num[i]) % 2 != 0:
        sumadeimpares = sumadeimpares + int(num[i])
    else:
        sumadepares = sumadepares + int(num[i])
resta = sumadepares-sumadeimpares
print('El resultado de la suma de los digitos es ',resta)
print('El resultado de la multiplicacion de los digitos es ',mult)
print('El resultado de la suma del string es ',sum)
print('El resultado de la division de la suma de los digitos entre el ultimo digito diferente de cero es ',div)

File: operaciones.js
const num="256";
let suma= 0;
let mult=1;
let resta = 0;
let sumadeimpares = 0;
let sumadepares = 0;
let divisor=1;
let d = 1;
for (let i = 0; i < num.length; i++) {
    suma= parseInt(num[i]) + suma;
    mult= parseInt(num[i]) * mult;
    if (parseInt(num[i]) % 2 != 0) {
        sumadeimpares = sumadeimpares + parseInt(num[i]);
    } else {
        sumadepares = sumadepares + parseInt(num[i]);
    }
    resta=  sumadepares-sumadeimpares
    if (parseInt(num.length-d==0)) {
        divisor = parseInt(num[i]);
        d = d-1;
    }else{
        divisor = parseInt(num[i])  
        div= suma / divisor;  
    }
}
console.log(resta);
console.log(suma);
console.log(mult);
console.log(div);
